That means the IDE interrupt (or the ST-DMA interrupt) doesn't have the
SA_SHIRQ flag set. Either allow for sharing of both interrupts, or just
ignore the request_irq failure in the IDE driver.

--- l/drivers/ide/ide-probe.c.org	2007-01-02 10:36:49.000000000 +0100
+++ l/drivers/ide/ide-probe.c	2006-10-21 15:24:58.000000000 +0200
@@ -1106,8 +1106,14 @@
 			/* clear nIEN */
 			hwif->OUTB(0x08, hwif->io_ports[IDE_CONTROL_OFFSET]);

-		if (request_irq(hwif->irq,&ide_intr,sa,hwif->name,hwgroup))
+		if (request_irq(hwif->irq,&ide_intr,sa,hwif->name,hwgroup)) {
+#if defined(__mc68000__) || defined(CONFIG_BLK_DEV_FALCON_IDE)
+			if (MACH_IS_ATARI && ATARIHW_PRESENT(IDE)) {
+				printk("ide-probe: not registered IDE interrupt for Falcon IDE!\n");
+			} else
+#endif
 	       		goto out_unlink;
+		}
 	}

 	/*

Signed-Off-By: schmitz@debian.org
